test/ci/shellcheck.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+shellcheck_args=(shellcheck --external-sources --shell bash)
+scriptDir=$(cd "${BASH_SOURCE[0]%/*}" && pwd)
+
+
+scheck_dir() {
+    local test_dir="$1"
+    echo "Shellchecking directory: ${test_dir}"
+
+    # Set working directory to `$test_dir` so the shellchek's
+    # `--external-sources` argument works better.
+    pushd "$test_dir"
+        find './' -type f -name '*.sh' -print0 | xargs -0 -I '{}' sh -c "echo 'Testing: {}'; ${shellcheck_args[*]} '{}'"
+    popd
```

Comment:
nit: can you redirect the output of `pushd` and `popd` to `/dev/null`? This makes the output of `shellcheck.sh` much more comprehensible.